### ISSUE TITLE
Allow a single asset to contain multiple data

### ISSUE THREAD

**Issue by RiscadoA** [GameDevTecnico/cubos#1274]
# Problem
Our asset flow lacks support for non-trivial asset handling.

The following scenario is a good example of this:
- UI Image elements use a render device (GPU) texture.
- That render device texture is created by sending an image to the GPU/render device.
- The image can be loaded from a file.

Currently, we can tackle this by providing an `Image` asset type, which loads a `.png` or equivalent.
Then, on the UI side, we keep a `Asset<Image>` field on a component, and the UI code must manually manage the render device texture creation.
This is cumbersome and error prone, and this is only one of the examples - we also have a similar situation with voxel grids.

# Solution
A single asset (UUID) should be able to contain multiple data types.
For example, taking the image example once again, we could have two asset types: `Image`, and `Texture`.
The UI would store an `Asset<Texture>`, keeping a reference to only the GPU/render device texture asset.
When loading texture data for a given asset, the `TextureBridge` load method is called, which internally loads the `Image` data of the same asset, and creates a render device texture.
As no reference is kept to the `Image` data, it can be unloaded, keeping only the `Texture` part.

To sum it up, we should:
- Identify asset data through their type and UUID, not only by their UUID.
- Change asset meta files to not contain the data extension (i.e., instead of `image.png.meta`, just `image.meta`).

**Comment by RiscadoA** [comment]
After thinking about this a bit further, I came to the conclusion that this might not be such a good idea, as multiple 'source' assets may need to be exported multiple times, with different options. For example, it should be possible to create multiple font atlas assets from a single font asset.

**Comment by RiscadoA** [comment]
Following up on the previous comment, we decided to keep the existing system, and simply have assets which don't have 'data file' counterparts. For example, we could have a `.mesh` asset which only contains a meta file, i.e., `car.mesh.meta`, but no `car.mesh` itself. Then, on the meta file, it would contain the UUID of the `.grd` from which the mesh is loaded.
